### DISCUSSION TITLE
Scripts to convert VASP OUTCAR to Allegro input

### DISCUSSION THREAD

**Discussion by Babuji-ph21d300** [mir-group/allegro#61] [Q&A] [answered]
Dear Team,

Thanks a lot for the Allegro software and making it opensource. I recently came across your  article on Allegro while looking for machine learned potentials (MLP). I had the experience of using invariant MLPs (e.g. [AENET](http://ann.atomistic.net/documentation/)).  I came across some [scripts](https://lipai.github.io/scripts/ml_scripts/outcar2xsf.html) on the web that convert OUTCAR to input data for aenet. I want to know if such scripts are available to directly convert OUTCAR to allegro input

I need some help in converting the VASP output files like OUTCAR to Allegro input files. I saw extensions like .extxyz and .npz and I don't know how these are different from simple .xyz extension files for atomic positions and lattice.

In my naive understanding, I'm assuming all we need a folder where

1.  OUTCAR file which will be converted to allegro input file(s)
2.  A '.yaml' file mentioning our requirements in the model training etc..
3.  A submission script (in case of a compute cluster) to a GPU, containing a path to allegro location

After the job finishes, the Allegro model generates a .pth file which can directly used as a forcefield in lammps (provided allegro and lammps are patched)  and run NVT simulations or any other simulations of interest.

I have seen the silicon trajectory file shared in collab, which has other data than coordinates and total energy, like lattice, properties, stress and free energies. Can I just get away wid the lattice and  atomic positions as in a XDATCAR/.xyz file, along with energy mentioned at each time step? 

If yes, is there any script for converting the OUTCAR to the .xyz ? 

Being new to this, I see that there are lots of gaps in my understanding of the allegro project and its application. Thanks a lot for your time and I highly appreciate any leads/suggestions. Have a nice day : )

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @Babuji-ph21d300 ,

Thanks for your interest in Allegro!

Yes, your description of the workflow sounds exactly right.  Your submission script, adapted to your cluster, will run the appropriate `nequip-train` command.

Regarding the OUTCAR to extXYZ conversion, please see example and discussion here: https://gist.github.com/simonbatzner/c2b05d38789b67f6fe5d3c75a4f2223d
